src/extensions/spikeamplitudes/SpikeAmplitudesView/SpikeAmplitudesPanel.ts

```diff
@@ -0,0 +1,155 @@
+import { Brush, CanvasPainter, Font, Pen } from "../../CanvasWidget/CanvasPainter"
+import { HitherContext, Recording, Sorting } from "../../extensionInterface"
+
+// const calculationPool = createCalculationPool({maxSimultaneous: 6})
+
+const colorList = [
+    'blue',
+    'green',
+    'red',
+    'orange',
+    'purple'
+]
+const colorForUnitId = (unitId: number) => {
+    while (unitId < 0) unitId += colorList.length
+    return colorList[unitId % colorList.length]
+} 
+
+class SpikeAmplitudesPanel {
+    _updateHandler: (() => void) | null = null
+    _timeRange: {min: number, max: number} | null = null
+    _timepoints: number[] | null = null
+    _amplitudes: number[] | null = null
+    _maxAmplitude: number | null = null
+    _calculationScheduled: boolean = false
+    _calculationError: Error | null = null
+    _yrange: {min: number, max: number} | null = null
+    _panelGroup: SpikeAmplitudesPanel[] | null = null
+    _includeZero = true
+    constructor(private args: {recording: Recording, sorting: Sorting, unitId: number, hither: HitherContext}) {
+    }
+    setTimeRange(timeRange: {min: number, max: number}) {
+        this._timeRange = timeRange
+    }
+    paint(painter: CanvasPainter, completenessFactor: number) {
+        const timeRange = this._timeRange
+        if (!timeRange) return
+        const font: Font = {pixelSize: 12, family: 'Arial'}
+        const color = colorForUnitId(this.args.unitId)
```

Comment:
I'm a little torn on this, because while it's good to have consistent coloring for each unit id, it'd also be pretty easy to wind up with a lot of unit IDs that are rendered with the same color, even though others are available.

Something to worry about for the future.